node-test/src/test/scala/org/bitcoins/node/NeutrinoNodeTest.scala

```diff
@@ -0,0 +1,153 @@
+        _ <- hashF
+        sync <- node.sync()
+      } yield sync
+
+      syncF.flatMap { _ =>
+        NodeTestUtil
+          .awaitSync(node, bitcoind)
+          .map(_ => succeed)
+      }
+  }
+
+  it must "stay in sync with a bitcoind instance" in {
+    nodeConnectedWithBitcoind: NeutrinoNodeConnectedWithBitcoind =>
+      val node = nodeConnectedWithBitcoind.node
+      val bitcoind = nodeConnectedWithBitcoind.bitcoind
+
+      //we need to generate 1 block for bitcoind to consider
+      //itself out of IBD. bitcoind will not sendheaders
+      //when it believes itself, or it's peer is in IBD
+      val gen1F =
+        bitcoind.getNewAddress.flatMap(bitcoind.generateToAddress(1, _))
+
+      //this needs to be called to get our peer to send us headers
+      //as they happen with the 'sendheaders' message
+      //both our spv node and our bitcoind node _should_ both be at the genesis block (regtest)
+      //at this point so no actual syncing is happening
+      val initSyncF = gen1F.flatMap { hashes =>
+        val syncF = node.sync()
+        for {
+          _ <- syncF
+          _ <- NodeTestUtil.awaitBestHash(hashes.head, node)
+        } yield ()
+      }
+
+      //start generating a block every 10 seconds with bitcoind
+      //this should result in 5 blocks
+      val startGenF = initSyncF.map { _ =>
+        //generate a block every 5 seconds
+        //until we have generated 5 total blocks
+        genBlockInterval(bitcoind)
+      }
+
+      startGenF.flatMap { _ =>
+        //we should expect 5 headers have been announced to us via
+        //the send headers message.
+        def has6BlocksF =
+          RpcUtil.retryUntilSatisfiedF(conditionF = () =>
+                                         node
+                                           .chainApiFromDb()
+                                           .flatMap(_.getBlockCount.map { c =>
+                                             c == 6
+                                           }),
+                                       duration = 1000.millis)
+
+        def has6FilterHeadersF =
+          RpcUtil.retryUntilSatisfiedF(
+            conditionF = () =>
+              node
+                .chainApiFromDb()
+                .flatMap(_.getHighestFilterHeader.map {
+                  header: Option[CompactFilterHeaderDb] =>
+                    header.exists(_.height == 6)
+                }),
+            duration = 1000.millis
+          )
+
+        def has6FiltersF =
+          RpcUtil.retryUntilSatisfiedF(conditionF = () =>
+                                         node
+                                           .chainApiFromDb()
+                                           .flatMap(_.getHighestFilter.map {
+                                             filter =>
+                                               filter.exists(_.height == 6)
+                                           }),
+                                       duration = 1000.millis)
```

Comment:
nit: I think that all these `retryUntilSatisfiedF` calls would look nicer if we used `{}` rather than `()` and then put the insides on the next line so indentation was nicer